File: 2017/Python/src/13.py
from AoC_tools import read_input, chinese_remainder


input = [[int(i) for i in item.split(': ')] for item in read_input().splitlines()]
print(input)

severity=0
for layer, depth in input : 
    if layer % (2 * (depth - 1)) == 0 : severity += depth * layer
print(severity)

delay=1
while(any([(delay+layer) % (2 * (depth - 1)) == 0  for layer, depth in input])):
    delay += 1

print(delay)

File: Events/2017/13/13.rs
aoc::main!();

fn preprocessing(input_: &str) -> Vec<(u32, u32)> {
    input_
    .lines()
    .map(|layer|
        layer
        .split(": ")
        .map(|n| n.parse::<u32>().unwrap())
        .collect_tuple()
        .map(|(layer, depth)| (layer, 2 * (depth - 1)))
        .unwrap())
    .collect_vec()
}

fn part_1(record: Vec<(u32, u32)>) -> u32 {
    record
    .iter()
    .map(|(layer, depth)|
        if layer % depth == 0 {layer * (depth / 2 + 1)}
        else {0})
    .sum()
}

fn part_2(record: Vec<(u32, u32)>) -> u32 {
    (1..)
    .find(|&delay| 
        record
        .iter()
        .all(|(layer, depth)| ((delay + layer) % depth) != 0))
    .unwrap()
}